lib/shopify_api/resources/fulfillment_order.rb

```diff
@@ -12,5 +12,41 @@ def fulfillments(options = {})
       fo_fulfillments = get(:fulfillments, options)
       fo_fulfillments.map { |fof| FulfillmentOrderFulfillment.new(fof.as_json) }
     end
+
+    def move(new_location_id:)
+      body = { fulfillment_order: { new_location_id: new_location_id } }.to_json
+      keyed_fos = load_keyed_attributes_from_response(post(:move, {}, body))
+      if keyed_fos&.fetch('original_fulfillment_order', nil)&.attributes
+        load(keyed_fos['original_fulfillment_order'].attributes, false, true)
+      end
+      keyed_fos
+    end
+
+    def cancel
+      keyed_fos = load_keyed_attributes_from_response(post(:cancel, {}, only_id))
+      if keyed_fos&.fetch('fulfillment_order', nil)&.attributes
+        load(keyed_fos['fulfillment_order'].attributes, false, true)
+      end
+      keyed_fos
+    end
+
+    def close
+      load_attributes_from_response(post(:close, {}, only_id))
+    end
+
+    private
+
+    def load_keyed_attributes_from_response(response)
+      return load_attributes_from_response(response) if response.code != '200'
+
+      keyed_fulfillments = ActiveSupport::JSON.decode(response.body)
```

Comment:
nit: correct me if I'm mistaken but this variable is called `keyed_fulfillments` but when you apply the map on it below you get a `key` and `fo_attributes` should it be `keyed_fulfillment_orders` instead ?

Also I think the structure returned here would look like :

```
{
          original_fulfillment_order: 
          moved_fulfillment_order: 
          remaining_fulfillment_order:
}
```

correct ?

Reply:
Yes, good catch of my copy/pasta fail